echo-pipelinetriggers/src/main/java/com/netflix/spinnaker/echo/pipelinetriggers/monitor/DockerEventMonitor.java

```diff
@@ -100,15 +100,22 @@ private boolean matchTags(String suppliedTag, String incomingTag) {
 
   @Override
   protected Predicate<Trigger> matchTriggerFor(final TriggerEvent event) {
-    DockerEvent dockerEvent = (DockerEvent) event;
+    return trigger -> isMatchingTrigger((DockerEvent)event, trigger);
+  }
+
+  private boolean isMatchingTrigger(DockerEvent dockerEvent, Trigger trigger) {
     String account = dockerEvent.getContent().getAccount();
     String repository = dockerEvent.getContent().getRepository();
-    String tag = dockerEvent.getContent().getTag();
-    return trigger -> trigger.getType().equals(TRIGGER_TYPE) &&
-      trigger.getRepository().equals(repository) &&
-      trigger.getAccount().equals(account) &&
-      ((trigger.getTag() == null && !tag.equals("latest"))
-        || trigger.getTag() != null && matchTags(trigger.getTag(), tag));
+    String eventTag = dockerEvent.getContent().getTag();
+    String triggerTagPattern = null;
+    if(trigger.getTag() != null && trigger.getTag().trim().length() > 0) {
```

Comment:
You can replace this if clause with `StringUtils.isEmpty()` from apache.lang. Also please add a space before the opening `(`.

Reply:
Actually this would be equivalent to `!StringUtils.isBlank()`.

Reply:
SGTM.